### PR TITLE
Fixes #20216 - Add logic to proxy manifest refresh

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

**Comment by chris1984** [comment]
@johnpmitsch can I get a review please.

File: app/lib/katello/resources/candlepin.rb [hunk 1]
```diff
@@ -295,9 +295,9 @@ def self.resource(url, client_cert, client_key, ca_file)
 
             uri.scheme = URI.parse(proxy_config[:host]).scheme
             uri.host = URI.parse(proxy_config[:host]).host
-            uri.port = proxy_config[:port].to_s
-            uri.user = proxy_config[:user].to_s
-            uri.password = proxy_config[:password].to_s
+            uri.port = proxy_config[:port].to_s if proxy_config[:port]
+            uri.user = proxy_config[:user].to_s if proxy_config[:user]
+            uri.password = proxy_config[:password].to_s if proxy_config[:password]
```
Comment (jlsherrill) [inline app/lib/katello/resources/candlepin.rb]: nitpick, you can just do:
```
uri.port = proxy_config[:port].try(:to_s)
uri.user = proxy_config[:user].try(:to_s)
uri.password = proxy_config[:password].try(:to_s)
```

**Comment by chris1984** [comment]
@jlsherrill fixed the nitpick can you review again please. John did a function test on it and shows its working for him.

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
Thanks @chris1984 !

**Pull request merged**